### ISSUE TITLE
volumeClaimTemplates only works with name: data-volume

### ISSUE THREAD

**Issue by fischerscode** [mongodb/mongodb-kubernetes-operator#177]
**Thanks for your great work!**

In release v0.2.0 the `spec.statefulSet.spec.volumeClaimTemplates.metadata.name` must be `data-volume`, otherwise two claims are created.

### How to reproduce:

1. `git clone https://github.com/mongodb/mongodb-kubernetes-operator.git #clone repo`
2. `git checkout v0.2.0 #use v.0.2.0`
3. `kubectl create -f deploy/crds/mongodb.com_mongodb_crd.yaml #deploy crd`
4. `kubectl create -f deploy/ #deploy operator`
5. Create file:
```
apiVersion: mongodb.com/v1
kind: MongoDB
metadata:
  name: mongo-test
spec:
  members: 3
  type: ReplicaSet
  version: "4.2.6"
  persistent: true
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: my-user
      db: admin
      passwordSecretRef:
        name: my-user-password
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
  statefulSet:
    spec:
      volumeClaimTemplates:
        - metadata:
            name: data-volume-2
          spec:
            accessModes: [ "ReadWriteOnce", "ReadWriteMany" ]
            resources:
              requests:
                storage: 20Gi

---


apiVersion: v1
kind: Secret
metadata:
  name: my-user-password
type: Opaque
stringData:
  password: 58LObjiMpxcjP1sMDW
```
6. `kubectl apply -f test.yaml #deploy test`

### expected behavior:
The used PersistantVolumeClaims should be named: `data-volume-2-mongo-test-x`

### actual behavior:
For every pod, two PVCs are created.
```
kubectl get pvc
NAME                         STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-volume-2-mongo-test-0   Bound    pvc-d83db90b-5d2d-4b7a-9c63-849d0b2ad0cb   20Gi       RWO,RWX        standard       66s
data-volume-2-mongo-test-1   Bound    pvc-d63c3c8e-95fa-4b1f-9eeb-7de9a7936c08   20Gi       RWO,RWX        standard       38s
data-volume-2-mongo-test-2   Bound    pvc-ca003ee3-3ce6-450c-a752-3c0046d3b5f3   20Gi       RWO,RWX        standard       6s
data-volume-mongo-test-0     Bound    pvc-d23d1a73-27a4-4e47-bb64-cccdd1adf311   10G        RWO            standard       66s
data-volume-mongo-test-1     Bound    pvc-291fd348-262c-435f-beb3-047c882d40f8   10G        RWO            standard       38s
data-volume-mongo-test-2     Bound    pvc-0277ef48-faa3-478b-955f-676a6d7838bb   10G        RWO            standard       6s
```

**Comment by Brice187** [comment]
`accessModes: [ "ReadWriteOnce", "ReadWriteMany" ]` many you should only choose the desired one?

**Comment by chatton** [comment]
Hi @fischerscode this issue should be resolved by the latest operation release 0.2.1, please test it again with the new version. Closing this issue but please re-open if the problem isn't resolved.
